src/main/kotlin/com/exactpro/th2/http/client/Main.kt

```diff
@@ -124,12 +121,19 @@ fun run(
     val incomingSequence = createSequence()
     val outgoingSequence = createSequence()
 
-    val onRequest = { request: RawHttpRequest ->
-        messageRouter.send(request.toBatch(connectionId, outgoingSequence()), SECOND.toString())
+    val onRequest: (RawHttpRequest) -> Unit = { request: RawHttpRequest ->
+        val rawMessage = request.toRawMessage(connectionId, outgoingSequence())
+
+        messageRouter.send(request.toRawMessage(connectionId, outgoingSequence()).toBatch(), SECOND.toString())
+        eventRouter.storeEvent("Sent HTTP request", rootEventId, listOf(rawMessage.metadata.id))
```

Comment:
> rootEventId

We should use message's parent event id instead (if it has one ofc)

Reply:
seems logical